components/movementsensor/imuwit/imu.go

```diff
@@ -303,14 +301,8 @@ func (imu *wit) Close() error {
 	imu.logger.Debug("Closing wit motion imu")
 	imu.cancelFunc()
 	imu.activeBackgroundWorkers.Wait()
-
-	if imu.port != nil {
-		if err := imu.port.Close(); err != nil {
-			return err
-		}
-		imu.port = nil
-	}
-
+	// resource_manager's TryClose now closes input/output Readers by calling Close on its
+	// driver's Close functions
```

Comment:
[minor] fix my sentence structure and English here.

Reply:
I am not sure resource manager has the ability to reach a Reader 's Close since it's only owned by Wit. I think the culprit here is `defer utils.UncheckedErrorFunc(imu.port.Close)` ~line 207. Which get called when you cancel the context earlier in the close function
I think we can either remove the line or change it to something like this
``` 
utils.UncheckedErrorFunc(func() error {
if imu.port != nil {
		if err := imu.port.Close(); err != nil {
			return err
		}
		imu.port = nil
                return nil
	})
```

Reply:
Good catch! I saw resource_manager in my stack trace and didn't quite know what was going on. I'll change that logic and keep close as it is now.

Reply:
Addressed, and tested on hardware.